src/Fixer/ClassNotation/FinalStaticAccessFixer.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author ntzm
+ */
+final class FinalStaticAccessFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Converts `static` access to `self` access in final classes.',
```

Comment:
I do believe we shall add an reasoning why it's good to do this.

Like, we do have `summary`, but we are lacking `description`.

I can provide description why we shall use `static` in final classes.
If we have final class - both, static and self will work the same way. Then, if for any reason one is opening that class for modification - by removing the `final` modifier from class declaration - he will forget to change `self` to `static` in all places...
For that, I'm also not a big fan of adding it to local PHP CS Fixer's config

-----------------

For this reasoning, I'm giving the review "request changes".
This does not mean "we won't merge the PR, ever", but "we have open topic that deserves to be addressed, don't merge before that".

Reply:
Changing from `self` to the looser `static` and forgetting it at some spots is not a concern I share, in other words I would use this rule. However, I agree, improving documentation is a good idea, so please do so :)

@ntzm please update the docs and resolve the conflicts

Reply:
What about making this configurable to choose what one want to achieve? `self` or `static`.

Reply:
Why would one want to use `static` in a final class?

Reply:
>For that, I'm also not a big fan of adding it to local PHP CS Fixer's config

I removed it, reverted the changes, rebased and resolved the conflicts on top of master and squashed to one commit